core/src/main/java/org/apache/brooklyn/core/objs/proxy/InternalPolicyFactory.java

```diff
@@ -125,7 +124,6 @@ public InternalPolicyFactory(ManagementContextInternal managementContext) {
             for (Map.Entry<ConfigKey<?>, Object> entry : spec.getConfig().entrySet()) {
                 pol.config().set((ConfigKey)entry.getKey(), entry.getValue());
             }
-            ConfigConstraints.assertValid(pol);
```

Comment:
Ah, I see you've removed the `assertValid` here. Why? Is that because the policy/enricher won't have finished initialising until `setEntity` etc is called? Is that an indication of a poorly implemented policy? Or the right thing to do?